src/core/tsi/ssl_transport_security.cc

```diff
@@ -2239,6 +2252,19 @@ static tsi_result create_tsi_ssl_handshaker(
   impl->base.vtable = &handshaker_vtable;
   impl->factory_ref = tsi_ssl_handshaker_factory_ref(factory);
   *handshaker = &impl->base;
+
+  grpc_core::CustomPrivateKeySign* signature =
+      static_cast<grpc_core::CustomPrivateKeySign*>(
+          SSL_CTX_get_ex_data(SSL_get_SSL_CTX(impl->ssl),
```

Comment:
I think we need to split this up and do some validity checks

Take a look at http://google3/third_party/grpc/src/core/tsi/ssl_transport_security.cc;l=1036-1047;rcl=819606209 or elsewhere in this file